File: Data Structures & Algorithms/permutation-string/submission-0.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        if(s2.length() < s1.length()) return false;

        int left = 0;
        int right =0;
        vector<int> v1(26);
        vector<int> v2(26);

        for(int i=0;i<s1.size();i++){
            v1[s1[i]-'a']++;
        }
        
        while(right<s2.size()){
            v2[s2[right]-'a']++;
            if((right-left+1) > s1.length()){
                v2[s2[left]-'a']--;
                left++;
            }
            if(v1 == v2) return true;
            right++;
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        #so i know its already sorted least to highest
        # one valid solution
        #i know that index1 !=index2 
        left , right = 0 , len(numbers)-1

        while left<right :
            currSum = numbers[left] + numbers[right]
            if currSum > target:
                right-=1
            elif currSum < target:
                left+=1
            else:
                return [left+1,right+1]